### ISSUE TITLE
redb::error::TransactionError access

### ISSUE THREAD

**Issue by STPR** [cberner/redb#594]
Hello,
Thanks for redb which looks promising 👍

I'm trying to use redb with others crates like tokio, reqwest, ....
I'm using thiserror to treat all the errors but since redb v0.18.0, i'm having an issue with every functions that return a TransactionError (begin_write, begin_read, commit).
No problem with StorageError, DatabaseError or TableError.
```
type mismatch in function arguments
expected function signature `fn(redb::error::TransactionError) -> _`
```
I've tested the following code for thiserror with no success:
```rust
#[derive(thiserror::Error)]
pub enum CustomError {
    #[error("redb - Database transaction failed!")]
    DbTransactionError(redb::error::TransactionError),
}
```
Error:
```
module `error` is private
```
I've seen that since v0.18.0 you've improved errors to be more granular so i guess it's not an issue and this is normal but i'm struggling to find a solution...

**Comment by CraZySacX** [comment]
I'm experiencing the same.

Seems like it's just missing from the public api exports:
https://github.com/cberner/redb/blob/9f52274e752a938e5abb91949aea9e138e7e49eb/src/lib.rs#L16

**Comment by cberner** [comment]
Oops, ya I forgot to export it when I refactored the error types. I'll fix it shortly

**Comment by cberner** [comment]
This should be fixed in 0.20.0 now
